<!doctype html>



  


<html class="theme-next mist use-motion">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>



<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />












  <link href="/vendors/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css"/>




  <link href="//fonts.useso.com/css?family=Lato:300,400,700,400italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">



<link href="/vendors/font-awesome/css/font-awesome.min.css?v=4.4.0" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=0.5.0" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="Hexo, NexT" />








  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=0.5.0" />






<meta name="description" content="Breif introduction to 10xhttps://www.10xgenomics.com/technology/
[1] Zheng G X Y, Lau B T, Schnall-Levin M, et al. Haplotyping germline and cancer genomes with high-throughput linked-read sequencing[J">
<meta property="og:type" content="article">
<meta property="og:title" content="Notes on 10x Longranger">
<meta property="og:url" content="http://yoursite.com/2017/03/07/intro-10x/index.html">
<meta property="og:site_name" content="Paprika">
<meta property="og:description" content="Breif introduction to 10xhttps://www.10xgenomics.com/technology/
[1] Zheng G X Y, Lau B T, Schnall-Levin M, et al. Haplotyping germline and cancer genomes with high-throughput linked-read sequencing[J">
<meta property="og:updated_time" content="2017-03-10T03:21:41.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Notes on 10x Longranger">
<meta name="twitter:description" content="Breif introduction to 10xhttps://www.10xgenomics.com/technology/
[1] Zheng G X Y, Lau B T, Schnall-Levin M, et al. Haplotyping germline and cancer genomes with high-throughput linked-read sequencing[J">



<script type="text/javascript" id="hexo.configuration">
  var NexT = window.NexT || {};
  var CONFIG = {
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post"},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: 0,
      author: 'Author'
    }
  };
</script>

  <title> Notes on 10x Longranger | Paprika </title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="default">

  










  
  
    
  

  <div class="container one-collumn sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-meta ">
  

  <div class="custom-logo-site-title">
    <a href="/"  class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <span class="site-title">Paprika</span>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>
  <p class="site-subtitle"></p>
</div>

<div class="site-nav-toggle">
  <button>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
  </button>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-10x genomics">
          <a href="/2017/03/07/intro-10x/" rel="section">
            
            10x genomics
          </a>
        </li>
      
        
        <li class="menu-item menu-item-algorithm">
          <a href="/categories/Algorithm/" rel="section">
            
            algorithm
          </a>
        </li>
      
        
        <li class="menu-item menu-item-introduction">
          <a href="/categories/Introduction/" rel="section">
            
            introduction
          </a>
        </li>
      
        
        <li class="menu-item menu-item-developing logs">
          <a href="/categories/Developing-Logs/" rel="section">
            
            developing logs
          </a>
        </li>
      
        
        <li class="menu-item menu-item-programming skills">
          <a href="/categories/Programming-Skills/" rel="section">
            
            programming skills
          </a>
        </li>
      
        
        <li class="menu-item menu-item-machine learning">
          <a href="/categories/Machine-Learning/" rel="section">
            
            machine learning
          </a>
        </li>
      
        
        <li class="menu-item menu-item-math">
          <a href="/categories/Math/" rel="section">
            
            math
          </a>
        </li>
      
        
        <li class="menu-item menu-item-statistic">
          <a href="/categories/Statistic/" rel="section">
            
            statistic
          </a>
        </li>
      
        
        <li class="menu-item menu-item-python">
          <a href="/tags/Python/" rel="section">
            
            python
          </a>
        </li>
      

      
    </ul>
  

  
</nav>

 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                Notes on 10x Longranger
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">Posted on</span>
            <time itemprop="dateCreated" datetime="2017-03-07T16:46:25+08:00" content="2017-03-07">
              2017-03-07
            </time>
          </span>

          

          
            
          

          

          
          

        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        <h1 id="Breif_introduction_to_10x"><a href="#Breif_introduction_to_10x" class="headerlink" title="Breif introduction to 10x"></a>Breif introduction to 10x</h1><p><a href="https://www.10xgenomics.com/technology/" target="_blank" rel="external">https://www.10xgenomics.com/technology/</a></p>
<p>[1] Zheng G X Y, Lau B T, Schnall-Levin M, et al. Haplotyping germline and cancer genomes with high-throughput linked-read sequencing[J]. Nature biotechnology, 2016.</p>
<p>[2] Mostovoy Y, Levy-Sakin M, Lam J, et al. A hybrid approach for de novo human genome sequence assembly and phasing[J]. Nature methods, 2016.</p>
<p>Refereces for these section:</p>
<ul>
<li><a href="http://www.novogene.com/index.php?controller=site&amp;action=news_view&amp;id=278" target="_blank" rel="external">Novogene: 10X genomics linked reads 解读(Chinese)</a></li>
<li><a href="https://zhuanlan.zhihu.com/p/20723934" target="_blank" rel="external">知乎解读(Chinese)</a></li>
</ul>
<h1 id="Running_longranger"><a href="#Running_longranger" class="headerlink" title="Running longranger"></a>Running <code>longranger</code></h1><h2 id="Initial_sitecheck"><a href="#Initial_sitecheck" class="headerlink" title="Initial sitecheck"></a>Initial sitecheck</h2><p>We recommend running the sitecheck so you can have an idea about whether the system you have meets the minimum requirements for the 10x software.</p>
<p>Different parts of the 10x suite have different requirements, so it would be helpful to know what you are planning on running.</p>
<p>The minimum requirements for the different packages can be found by clicking: <a href="https://support.10xgenomics.com/genome-exome/software/overview/latest/system-requirements" target="_blank" rel="external">Longranger</a> <a href="https://support.10xgenomics.com/single-cell/software/overview/system-requirements" target="_blank" rel="external">Cellranger</a> <a href="https://support.10xgenomics.com/de-novo-assembly/software/overview/latest/system-requirements" target="_blank" rel="external">Supernova</a></p>
<h3 id="Open_file_limit"><a href="#Open_file_limit" class="headerlink" title="Open file limit"></a>Open file limit</h3><p>The required “global open file limit” for running <code>longranger</code> is at least 131072. The default open file limit for Linux user is 1024, you can use <code>ulimit -n</code> to check and reset.</p>
<figure class="highlight xquery"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="variable">$ulimit</span> -n</span><br><span class="line"><span class="number">1024</span></span><br><span class="line"></span><br><span class="line"><span class="variable">$ulimit</span> -n <span class="number">131072</span></span><br><span class="line"><span class="variable">$ulimit</span> -n</span><br><span class="line"><span class="number">13072</span></span><br></pre></td></tr></table></figure>
<p>This is only important if you are planning to run Longranger. It’s the only pipeline that has an actual need for that many, but because of the shared code base the other two check for it as well. To work around this with Cellranger or Supernova, you can provide the “–nopreflight” option when you run the respective pipelines. This will not work with Longranger because as I indicated, Longranger actually needs to open a very large number of files.</p>
<h3 id="System_Memory"><a href="#System_Memory" class="headerlink" title="System Memory"></a>System Memory</h3><p>The <code>longranger</code> pipelines are very CPU and memory intensive, they do require substantial computing power in order to run. For RAM,</p>
<figure class="highlight http"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="attribute">Longranger</span>: <span class="string">96 GB</span></span><br><span class="line"><span class="attribute">Cellranger</span>: <span class="string">64 GB</span></span><br><span class="line"><span class="attribute">Supernova</span>: <span class="string">384 GB</span></span><br></pre></td></tr></table></figure>
<p>To check the memory in your system, check <code>/proc/cpuinfo</code> and <code>/proc/meminfo</code>.</p>
<h2 id="longranger_wgs"><a href="#longranger_wgs" class="headerlink" title="longranger wgs"></a><code>longranger wgs</code></h2><h3 id="FASTQ_naming_convention_for_longranger_wgs"><a href="#FASTQ_naming_convention_for_longranger_wgs" class="headerlink" title="FASTQ naming convention for longranger wgs"></a>FASTQ naming convention for <code>longranger wgs</code></h3><p><code>longranger wgs</code> first does preflight check to see if there are valid fastq files lie in the specified path. <code>longranger</code> accepts two kinds of naming convention, called “10x preprocess” and “bcl2fastq demultiplex”</p>
<ol>
<li>“10x preprocess” means the fastq data are made directly by <code>longranger mkfastq</code>, e.g. The NA12787 sample data provided by 10x</li>
</ol>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">read-I1_si-AAGGGTGA_lane-<span class="number">001</span>-chunk-<span class="number">0002</span><span class="class">.fastq</span><span class="class">.gz</span></span><br><span class="line">read-RA_si-AAGGGTGA_lane-<span class="number">001</span>-chunk-<span class="number">0002</span><span class="class">.fastq</span><span class="class">.gz</span></span><br></pre></td></tr></table></figure>
<p>Customarily, there are I1 index files and RA interleaved read files.</p>
<ol>
<li>“bcl2fastq demultiplex” refers to those made by calling standard Illumia bcl2fastq process. Generally, there are I1 index files and R1, R2 read files. e.g.</li>
</ol>
<figure class="highlight stylus"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ATGCTGAA_S0_L004_I1_001<span class="class">.fastq</span><span class="class">.gz</span></span><br><span class="line">ATGCTGAA_S0_L004_R1_001<span class="class">.fastq</span><span class="class">.gz</span></span><br><span class="line">ATGCTGAA_S0_L004_R2_001<span class="class">.fastq</span><span class="class">.gz</span></span><br></pre></td></tr></table></figure>
<p>The accepted naming pattern is <code>&#39;%s_%s_%s_%s_%s.*&#39;% (sample_index,sample_id,lane,read_type,chunk)</code> where</p>
<ul>
<li><code>sample_index</code> the 8bp sample index for reads in that file</li>
<li><code>sample_id</code> customized, generally are S0, S1</li>
<li><code>lane = &#39;L%03d&#39; % lane_number</code> lane number</li>
<li><code>read_type</code> 10x read type can be I1, R1, R2</li>
<li><code>chunk = &#39;[0-9][0-9][0-9]&#39;</code> generally starts from 001</li>
<li><code>*</code> customized suffix, like <code>fastq.gz</code></li>
</ul>
<h3 id="wgs_options"><a href="#wgs_options" class="headerlink" title="wgs options"></a><code>wgs</code> options</h3><p>An example of <code>longranger wgs</code> is:</p>
<figure class="highlight haml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">longranger wgs --id=sample1 \  </span><br><span class="line">               -<span class="ruby">-reference=refdata-hg19-<span class="number">2.1</span>.<span class="number">0</span> \</span><br><span class="line"></span>               -<span class="ruby">-fastqs=sample1_data \</span><br><span class="line"></span>               -<span class="ruby">-sample=<span class="constant">ATGCTGAA</span>,<span class="constant">CCCACAGT</span>,<span class="constant">GGATGCCG</span>,<span class="constant">TATGATTC</span></span></span><br></pre></td></tr></table></figure>
<p>where</p>
<ul>
<li><code>id</code> is the customized folder name for you to tell <code>longranger</code> where to store all the analysis data and logs</li>
<li><code>sample</code> are the read sample index you want <code>longranger</code> to analysis, they must be overlapped with all your fastq files prefix</li>
</ul>
<h3 id="Use_I1_Files_or_Not"><a href="#Use_I1_Files_or_Not" class="headerlink" title="Use I1 Files or Not"></a>Use I1 Files or Not</h3><p>There is no difference in performance with or without the I1 files. This I1 sample index read is only used to demultiplex the samples so as long as your fastq files are already separated you don’t absolutely need it for Long Ranger.</p>
<h3 id="Put_FASTQ_for_One_Sample_Together"><a href="#Put_FASTQ_for_One_Sample_Together" class="headerlink" title="Put FASTQ for One Sample Together"></a>Put FASTQ for One Sample Together</h3><p>You should combine all the fastq files for a single sample (four sets) when running the analysis. If you don’t do this then the software will produce four separate outputs for each sample. The easiest way to do this is to create a separate directory for each sample, and then move all of the R1,R2 pairs for a single sample into that directory. That way when you run <code>longranger wgs</code> you can just point it to the single directory and have it analyze the entire sample.</p>
<h3 id="Deal_with_Chunks"><a href="#Deal_with_Chunks" class="headerlink" title="Deal with Chunks"></a>Deal with Chunks</h3><p>The first stage of the pipeline (fastq splitting) will run more efficiently if the data is in “chunks”.  The fastq processing is single threaded so the more “chunks” there are the better the pipeline can utilize multi-core systems.  It honestly doesn’t take that much time to process the fastq files though, so I wouldn’t worry about splitting them up.  My recommendation would be to just run the pipeline on the intact files.  After the first stage of the pipeline is complete the data will be automatically chunked for all the rest of the steps.</p>
<h1 id="Beyond_longranger_command"><a href="#Beyond_longranger_command" class="headerlink" title="Beyond longranger command"></a>Beyond <code>longranger</code> command</h1><h2 id="Sample_index"><a href="#Sample_index" class="headerlink" title="Sample index"></a>Sample index</h2><p>The sample index ( 8 base pairs) is used to multiplex (mix multiple samples) on the flow cell. Each 10x library has a 10x sample index. That index is made up for 4 actual sample oligos. Click to find more on 10x website: <a href="https://support.10xgenomics.com/genome-exome/sequencing/doc/specifications-sequencing-requirements-for-genome-and-exome" target="_blank" rel="external">Sample Index Sets for Genome &amp; Exome</a>, <a href="https://support.10xgenomics.com/genome-exome/sequencing/doc/specifications-sample-index-sets-for-genome-and-exome" target="_blank" rel="external">Sequencing Requirements for Genome &amp; Exome</a></p>
<p>The collection of sample indices can be found inside of <code>longranger-2.1.2</code></p>
<figure class="highlight crystal"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">longranger-<span class="number">2.1</span>.<span class="number">2</span>/longranger-cs/<span class="number">2.1</span>.<span class="number">2</span>/tenkit/<span class="class"><span class="keyword">lib</span>/<span class="title">python</span>/<span class="title">tenkit</span>/<span class="title">barcodes</span>/<span class="title">constants</span>.<span class="title">py</span></span></span><br></pre></td></tr></table></figure>
<h2 id="Barcode"><a href="#Barcode" class="headerlink" title="Barcode"></a>Barcode</h2><p>The 10x barcode is a unique barcode on each GEM. For Chromium Genome, it is 16 base pairs long. You will expect to see 1Million+ of these. This is found on the first 16 bases of the R1 read.</p>
<p>Barcode whitelist is stored inside <code>4M-with-alts-february-2016.txt</code>, and it has 4,792,320 unique 16-mer barcodes. The path is</p>
<figure class="highlight crystal"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">longranger-<span class="number">2.1</span>.<span class="number">2</span>/longranger-cs/<span class="number">2.1</span>.<span class="number">2</span>/tenkit/<span class="class"><span class="keyword">lib</span>/<span class="title">python</span>/<span class="title">tenkit</span>/<span class="title">barcodes</span>/</span></span><br></pre></td></tr></table></figure>
<p>If you want to generate a version of the FASTQ files with the barcode stripped out and error corrected, you can run the <a href="https://support.10xgenomics.com/genome-exome/software/pipelines/latest/advanced/other-pipelines" target="_blank" rel="external">Long Ranger BASIC pipeline</a></p>
<h2 id="Read_Type"><a href="#Read_Type" class="headerlink" title="Read Type"></a>Read Type</h2><p>The read type in BCL <code>ReadInfo.xml</code> files are <code>R1, I1, I2, R2</code>, and the <code>bcl2fastq</code> output file naming are <code>R1, R2, R3, R4, I1</code></p>
<p>The mapping relationships claimed by 10x support are:</p>
<figure class="highlight armasm"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="label">R1</span>/<span class="literal">R2</span>/<span class="literal">R3</span>/<span class="literal">R4</span> are for the <span class="literal">R1</span>,<span class="literal">R2</span>, i7 index <span class="keyword">and </span>i5 index.</span><br><span class="line"><span class="label">Some</span> of our older Gemcode <span class="keyword">and </span>Single Cell <span class="literal">v1</span> chemistries</span><br><span class="line"><span class="label">leveraged</span> the  index sites to store more than sample indices.</span><br><span class="line"><span class="label">These</span> included <span class="number">10</span>x <span class="keyword">barcode </span>or UMI<span class="string">'s.</span></span><br></pre></td></tr></table></figure>
<h2 id="R1_2C_R2_2C_and_RA"><a href="#R1_2C_R2_2C_and_RA" class="headerlink" title="R1, R2, and RA"></a>R1, R2, and RA</h2><p>For 151bp R1 read, there is only <code>151-16-6-1=128</code> base pairs left for downstream analysis after running <code>longranger wgs</code>. Here is the 10x support statement:</p>
<figure class="highlight livecodeserver"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">The <span class="keyword">first</span> <span class="number">16</span> bases are <span class="operator">the</span> <span class="number">10</span>x barcode.</span><br><span class="line">The next <span class="number">6</span> are n-mer sequencers</span><br><span class="line">(<span class="keyword">for</span> binding <span class="built_in">to</span> different parts <span class="operator">of</span> <span class="operator">the</span> molecule)</span><br><span class="line">The next <span class="number">1</span> base is typically always poor quality</span><br><span class="line">due <span class="built_in">to</span> <span class="operator">the</span> n-mer. We trim that off.</span><br></pre></td></tr></table></figure>
<p>The R2 read are just normal read, and RA read are interleaved fastq files.</p>
<h1 id="Code_Structure"><a href="#Code_Structure" class="headerlink" title="Code Structure"></a>Code Structure</h1>
      
    </div>

    <div>
      
        
      
    </div>

    <footer class="post-footer">
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2016/06/13/log-django/" rel="next" title="log_django">
                <i class="fa fa-chevron-left"></i> log_django
              </a>
            
          </div>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2017/04/13/bioinfo-bwa/" rel="prev" title="bioinfo-bwa">
                bioinfo-bwa <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap" >
            Table of Contents
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            Overview
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel ">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/figures/me.png"
               alt="Lingxi Chen" />
          <p class="site-author-name" itemprop="name">Lingxi Chen</p>
          <p class="site-description motion-element" itemprop="description"></p>
        </div>
        <nav class="site-state motion-element">
          <div class="site-state-item site-state-posts">
            <a href="/">
              <span class="site-state-item-count">40</span>
              <span class="site-state-item-name">posts</span>
            </a>
          </div>
          
          
            <div class="site-state-item site-state-categories">
              
                <span class="site-state-item-count">12</span>
                <span class="site-state-item-name">categories</span>
              
            </div>
          

          
            <div class="site-state-item site-state-tags">
              
                <span class="site-state-item-count">21</span>
                <span class="site-state-item-name">tags</span>
              
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
        </div>

        
        

        <div class="links-of-author motion-element">
          
        </div>

      </section>

      
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc-indicator-top post-toc-indicator">
            <i class="fa fa-angle-double-up"></i>
          </div>
          <div class="post-toc">
            
              
            
            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#Breif_introduction_to_10x"><span class="nav-number">1.</span> <span class="nav-text">Breif introduction to 10x</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Running_longranger"><span class="nav-number">2.</span> <span class="nav-text">Running longranger</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Initial_sitecheck"><span class="nav-number">2.1.</span> <span class="nav-text">Initial sitecheck</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Open_file_limit"><span class="nav-number">2.1.1.</span> <span class="nav-text">Open file limit</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#System_Memory"><span class="nav-number">2.1.2.</span> <span class="nav-text">System Memory</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#longranger_wgs"><span class="nav-number">2.2.</span> <span class="nav-text">longranger wgs</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#FASTQ_naming_convention_for_longranger_wgs"><span class="nav-number">2.2.1.</span> <span class="nav-text">FASTQ naming convention for longranger wgs</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#wgs_options"><span class="nav-number">2.2.2.</span> <span class="nav-text">wgs options</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Use_I1_Files_or_Not"><span class="nav-number">2.2.3.</span> <span class="nav-text">Use I1 Files or Not</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Put_FASTQ_for_One_Sample_Together"><span class="nav-number">2.2.4.</span> <span class="nav-text">Put FASTQ for One Sample Together</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Deal_with_Chunks"><span class="nav-number">2.2.5.</span> <span class="nav-text">Deal with Chunks</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Beyond_longranger_command"><span class="nav-number">3.</span> <span class="nav-text">Beyond longranger command</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Sample_index"><span class="nav-number">3.1.</span> <span class="nav-text">Sample index</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Barcode"><span class="nav-number">3.2.</span> <span class="nav-text">Barcode</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Read_Type"><span class="nav-number">3.3.</span> <span class="nav-text">Read Type</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#R1_2C_R2_2C_and_RA"><span class="nav-number">3.4.</span> <span class="nav-text">R1, R2, and RA</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Code_Structure"><span class="nav-number">4.</span> <span class="nav-text">Code Structure</span></a></li></ol></div>
            
          </div>
          <div class="post-toc-indicator-bottom post-toc-indicator">
            <i class="fa fa-angle-double-down"></i>
          </div>
        </section>
      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2017</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Lingxi Chen</span>
</div>

<div class="powered-by">
  Powered by <a class="theme-link" href="http://hexo.io">Hexo</a>
</div>

<div class="theme-info">
  Theme -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>



      </div>
    </footer>

    <div class="back-to-top">
      <i class="fa fa-arrow-up"></i>
    </div>
  </div>

  


  




<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>


  <script type="text/javascript" src="/vendors/jquery/index.js?v=2.1.3"></script>

  <script type="text/javascript" src="/vendors/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  <script type="text/javascript" src="/vendors/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  <script type="text/javascript" src="/vendors/velocity/velocity.min.js"></script>

  <script type="text/javascript" src="/vendors/velocity/velocity.ui.min.js"></script>

  <script type="text/javascript" src="/vendors/fancybox/source/jquery.fancybox.pack.js"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=0.5.0"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=0.5.0"></script>



  
  

  
  
<script type="text/javascript" src="/js/src/scrollspy.js?v=0.5.0"></script>

<script type="text/javascript" id="sidebar.toc.highlight">
  $(document).ready(function () {
    var tocSelector = '.post-toc';
    var $tocSelector = $(tocSelector);
    var activeCurrentSelector = '.active-current';

    $tocSelector
      .on('activate.bs.scrollspy', function () {
        var $currentActiveElement = $(tocSelector + ' .active').last();

        removeCurrentActiveClass();
        $currentActiveElement.addClass('active-current');

        $tocSelector[0].scrollTop = $currentActiveElement.position().top;
      })
      .on('clear.bs.scrollspy', function () {
        removeCurrentActiveClass();
      });

    function removeCurrentActiveClass () {
      $(tocSelector + ' ' + activeCurrentSelector)
        .removeClass(activeCurrentSelector.substring(1));
    }

    function processTOC () {
      getTOCMaxHeight();
      toggleTOCOverflowIndicators();
    }

    function getTOCMaxHeight () {
      var height = $('.sidebar').height() -
                   $tocSelector.position().top -
                   $('.post-toc-indicator-bottom').height();

      $tocSelector.css('height', height);

      return height;
    }

    function toggleTOCOverflowIndicators () {
      tocOverflowIndicator(
        '.post-toc-indicator-top',
        $tocSelector.scrollTop() > 0 ? 'show' : 'hide'
      );

      tocOverflowIndicator(
        '.post-toc-indicator-bottom',
        $tocSelector.scrollTop() >= $tocSelector.find('ol').height() - $tocSelector.height() ? 'hide' : 'show'
      )
    }

    $(document).on('sidebar.motion.complete', function () {
      processTOC();
    });

    $('body').scrollspy({ target: tocSelector });
    $(window).on('resize', function () {
      if ( $('.sidebar').hasClass('sidebar-active') ) {
        processTOC();
      }
    });

    onScroll($tocSelector);

    function onScroll (element) {
      element.on('mousewheel DOMMouseScroll', function (event) {
          var oe = event.originalEvent;
          var delta = oe.wheelDelta || -oe.detail;

          this.scrollTop += ( delta < 0 ? 1 : -1 ) * 30;
          event.preventDefault();

          toggleTOCOverflowIndicators();
      });
    }

    function tocOverflowIndicator (indicator, action) {
      var $indicator = $(indicator);
      var opacity = action === 'show' ? 1 : 0;
      $indicator.velocity ?
        $indicator.velocity('stop').velocity({
          opacity: opacity
        }, { duration: 100 }) :
        $indicator.stop().animate({
          opacity: opacity
        }, 100);
    }

  });
</script>

<script type="text/javascript" id="sidebar.nav">
  $(document).ready(function () {
    var html = $('html');
    var TAB_ANIMATE_DURATION = 200;
    var hasVelocity = $.isFunction(html.velocity);

    $('.sidebar-nav li').on('click', function () {
      var item = $(this);
      var activeTabClassName = 'sidebar-nav-active';
      var activePanelClassName = 'sidebar-panel-active';
      if (item.hasClass(activeTabClassName)) {
        return;
      }

      var currentTarget = $('.' + activePanelClassName);
      var target = $('.' + item.data('target'));

      hasVelocity ?
        currentTarget.velocity('transition.slideUpOut', TAB_ANIMATE_DURATION, function () {
          target
            .velocity('stop')
            .velocity('transition.slideDownIn', TAB_ANIMATE_DURATION)
            .addClass(activePanelClassName);
        }) :
        currentTarget.animate({ opacity: 0 }, TAB_ANIMATE_DURATION, function () {
          currentTarget.hide();
          target
            .stop()
            .css({'opacity': 0, 'display': 'block'})
            .animate({ opacity: 1 }, TAB_ANIMATE_DURATION, function () {
              currentTarget.removeClass(activePanelClassName);
              target.addClass(activePanelClassName);
            });
        });

      item.siblings().removeClass(activeTabClassName);
      item.addClass(activeTabClassName);
    });

    $('.post-toc a').on('click', function (e) {
      e.preventDefault();
      var targetSelector = NexT.utils.escapeSelector(this.getAttribute('href'));
      var offset = $(targetSelector).offset().top;
      hasVelocity ?
        html.velocity('stop').velocity('scroll', {
          offset: offset  + 'px',
          mobileHA: false
        }) :
        $('html, body').stop().animate({
          scrollTop: offset
        }, 500);
    });

    // Expand sidebar on post detail page by default, when post has a toc.
    NexT.motion.middleWares.sidebar = function () {
      var $tocContent = $('.post-toc-content');

      if (CONFIG.scheme !== 'Pisces' && (CONFIG.sidebar.display === 'post' || CONFIG.sidebar.display === 'always')) {
        if ($tocContent.length > 0 && $tocContent.html().trim().length > 0) {
          NexT.utils.displaySidebar();
        }
      }
    };
  });
</script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=0.5.0"></script>



  



  



  
  
  
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"]  ],
        processEscapes: true,
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
  </script>

  <script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
      var all = MathJax.Hub.getAllJax(), i;
      for (i=0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
      }
    });
  </script>

  
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  


  


</body>
</html>
